### PR TITLE
[C#] Rename exercises

### PR DESCRIPTION
**Pull request by ErikSchierboom**
See https://github.com/exercism/v3/issues/2589

### CODE REVIEW

File: languages/csharp/config.json [hunk 1]
```diff
@@ -11,43 +11,50 @@
...
+        "slug": "bird-watcher",
+        "name": "Bird Watcher",
         "uuid": "b6c532c9-1e89-4fbf-8f08-27f5befb5bb8",
         "concepts": ["arrays", "for-loops", "foreach-loops"],
         "prerequisites": ["booleans", "classes", "if-statements"]
       },
       {
-        "slug": "basics",
+        "slug": "lucians-luscious-lasagna",
+        "name": "Lucians Luscious Lasagna",
```
Comment (iHiD) [inline languages/csharp/config.json:22]:
```suggestion
        "name": "Lucian's Luscious Lasagna",
```

File: languages/csharp/config.json [hunk 2]
```diff
@@ -11,43 +11,50 @@
...
         "uuid": "7d358894-4fbd-4c91-b49f-d68f1c5aa6bc",
         "concepts": ["basics"],
         "prerequisites": []
       },
       {
-        "slug": "booleans",
+        "slug": "annalyns-infiltration",
+        "name": "Annalyns Infiltration",
```
Comment (iHiD) [inline languages/csharp/config.json:29]:
```suggestion
        "name": "Annalyn's Infiltration",
```

File: languages/csharp/config.json [hunk 3]
```diff
@@ -11,43 +11,50 @@
...
+        "slug": "squeaky-clean",
+        "name": "Squeaky Clean",
         "uuid": "e74c6cd6-add7-45d1-abcb-513418e9d2d0",
         "concepts": ["chars", "string-builder"],
         "prerequisites": ["if-statements", "for-loops", "strings"]
       },
       {
-        "slug": "classes",
+        "slug": "elons-toys",
+        "name": "Elons Toys",
```
Comment (iHiD) [inline languages/csharp/config.json:50]:
```suggestion
        "name": "Elon's Toys",
```

File: languages/csharp/config.json [hunk 4]
```diff
@@ -216,25 +239,29 @@
         ]
       },
       {
-        "slug": "nullability",
+        "slug": "hello-its-tim-from-marketing",
+        "name": "Hello Its Tim From Marketing",
```
Comment (iHiD) [inline languages/csharp/config.json:243]:
```suggestion
        "slug": "tim-from-marketing",
        "name": "Tim From Marketing",
```

File: languages/csharp/config.json [hunk 5]
```diff
@@ -216,25 +239,29 @@
...
-        "slug": "numbers",
+        "slug": "cars-assemble",
+        "name": "Cars Assemble",
         "uuid": "93fbc7cf-3a7e-4450-ad22-e30129c36bb9",
         "concepts": ["if-statements", "numbers"],
         "prerequisites": ["basics"]
       },
       {
-        "slug": "object-initializers",
+        "slug": "why-cant-i-push-to-master",
```
Comment (iHiD) [inline languages/csharp/config.json:256]:
```suggestion
        "slug": "why-cant-i-push-to-main",
```

File: languages/csharp/config.json [hunk 6]
```diff
@@ -216,25 +239,29 @@
...
+        "name": "Why Cant I Push To Master",
```
Comment (mikedamay) [inline languages/csharp/config.json:257]:
```suggestion
        "name": "Why Cant I Push To Main",
```

**Review by mikedamay — COMMENTED** [review]
Looks good to me.

Obviously @iHiD 's 2 name changes need to be applied to file names and configs.  My approach would be to process the automated changes and raise a separate PR or regenerate the automated stuff.

**Comment by ErikSchierboom** [comment]
I've updated the PR with the latest renames.

**Comment by ErikSchierboom** [comment]
@mikedamay I found a duplicate `exceptions` entry in the config.json BTW :)

**Review by mikedamay — APPROVED** [review]
LGTM (all 411 files).  Press a button and we will know for sure

**Comment by ErikSchierboom** [comment]
One follow-up step would be to check if the class names match what we expect from the updated exercise name.

**Comment by mikedamay** [comment]
Do you want to align the all class names with the file names in this PR?  That doesn't work to well for something like Tim From Marketing.

Weird.  I'm sure I left this (or similar) yesterday.  Hey ho.

**Comment by ErikSchierboom** [comment]
@mikedamay No, I think we should do a separate PR(or PRs?) for that. And it probably doesn't make sense everywhere. We should just check to see if there are places where it does.

**Comment by mikedamay** [comment]
Ok I'll take a look

**Pull request merged**